config/src/parameters/actual.rs

```diff
@@ -99,27 +99,18 @@ pub enum Genesis {
     },
     /// The peer is responsible for submitting the genesis block
     Full {
-        /// Genesis account key pair
-        key_pair: KeyPair,
-        /// Path to `RawGenesisBlock`
-        file: WithOrigin<PathBuf>,
+        /// Genesis account public key
+        public_key: PublicKey,
+        /// Path to `GenesisTransaction`
+        signed_file: WithOrigin<PathBuf>,
     },
 }
 
 impl Genesis {
     /// Access the public key, which is always present in the genesis config
     pub fn public_key(&self) -> &PublicKey {
         match self {
-            Self::Partial { public_key } => public_key,
-            Self::Full { key_pair, .. } => key_pair.public_key(),
-        }
-    }
-
-    /// Access the key pair, if present
-    pub fn key_pair(&self) -> Option<&KeyPair> {
-        match self {
-            Self::Partial { .. } => None,
-            Self::Full { key_pair, .. } => Some(key_pair),
+            Self::Partial { public_key } | Self::Full { public_key, .. } => public_key,
```

Comment:
`Partial` and `Full` variants don't make much sense anymore, and this would be cleaner & just the same type-safe:

```rust
struct Genesis {
  public_key: PublicKey,
  signed_file: Option<PathBuf>
}
```

I am fine with not touching it too.